harvest/api/kraken.py

```diff
@@ -0,0 +1,292 @@
+        return self.create_secret(path)
+
+    def setup(self, watch: List[str], interval: str, trader=None, trader_main=None):
+        self.watch_crypto = []
+        if is_crypto(s):
+            self.watch_crypto.append(s)
+        else:
+            warning("Kraken does not support stocks!")
+
+        self.option_cache = {}
+        super().setup(watch, interval, interval, trader, trader_main)
+
+    def exit(self):
+        self.option_cache = {}
+
+    def main(self):
+        df_dict = {}
+        df_dict.update(self.fetch_latest_crypto_price())
+
+        self.trader_main(df_dict)
+
+    @API._exception_handler
+    def fetch_latest_crypto_price(self):
+        dfs = {}
+        for symbol in self.watch_cryptos:
+            dfs[symbol] = self.fetch_price_history(symbol, self.interval, now() - dt.timedelta(days=7), now()).iloc[[0]]
+        return dfs
+
+    # -------------- Streamer methods -------------- #
+
+    @API._exception_handler
+    def fetch_price_history( self,
+        symbol: str,
+        interval: str,
+        start: dt.datetime = None,
+        end: dt.datetime = None,
+       ):
+
+        debug(f"Fetching {symbol} {interval} price history")
+
+        if start is None:
+            start = now() - dt.timedelta(days=365 * 5)
+        if end is None:
+            end = now()
+
+        if start >= end:
+            return pd.DataFrame()
+
+        if interval not in self.interval_list:
+            raise Exception(f"Interval {interval} not in interval list. Possible options are: {self.interval_list}")
+        val, unit = expand_interval(interval)
+        df = self.get_data_from_kraken(symbol, val, unit, start, end)
+
+        return df
+
+    @API._exception_handler
+    def fetch_chain_info(self, symbol: str):
+        return {
+            "id": "n/a",
+            "exp_dates": [str_to_date(s) for s in self.watch_ticker[symbol].options],
+            "multiplier": 100
+        }
+
+    @API._exception_handler
+    def fetch_chain_data(self, symbol: str, date: dt.datetime):
+
+        if bool(self.option_cache) and symbol in self.option_cache and date in self.option_cache[symbol]:
+            return self.option_cache[symbol][date]
+
+        df = pd.DataFrame(columns=["contractSymbol", "exp_date", "strike", "type"])
+
+        chain = self.watch_ticker[symbol].option_chain(date_to_str(date))
+        puts = chain.puts
+        puts['type'] = 'put'
+        calls = chain.calls
+        calls['type'] = 'call'
+        df = df.append(puts)
+        df = df.append(calls)
+
+        df = df.rename(columns={"contractSymbol": "occ_symbol"})
+        df['exp_date'] = df.apply(lambda x: self.occ_to_data(x['occ_symbol'])[1], axis=1)
+        df = df[["occ_symbol", "exp_date", "strike", "type"]]
+        df.set_index('occ_symbol', inplace=True)
+
+        if not symbol in self.option_cache:
+            self.option_cache[symbol] = {}
+        self.option_cache[symbol][date] = df
+
+        return df
```

Comment:
What is `fetch_chain_data` supposed to do?

Reply:
This function also is mainly for options.